### PR TITLE
Add support for PSR-15

### PR DESCRIPTION
**Pull request by flyinghail**
[PSR-15](https://www.php-fig.org/psr/psr-15/) contains two packages:

- [psr\http-server-handler](https://github.com/php-fig/http-server-handler)
- [psr\http-server-middleware](https://github.com/php-fig/http-server-middleware)

and require PHP >= 7.0

### CODE REVIEW

**Comment by sergeyklay** [comment]
@flyinghail Could you please explain a bit more why do we need bump PHP version

**Comment by flyinghail** [comment]
@sergeyklay 

- The new PSR stands all bump to PHP 7.0  ([PSR-15](https://www.php-fig.org/psr/psr-15/) and [PSR-18](https://github.com/php-fig/fig-standards/tree/master/proposed/http-client/)). There will be more and more PSR standards bump to PHP 7 in the future.

- [PHP Versions Stats - 2017.2 Edition](https://seld.be/notes/php-versions-stats-2017-2-edition) shows:

  - 36.63% people are on PHP 7.1 and growing
  - 30.76% on PHP 7.0

- [Majority of PHP frameworks start to require PHP7.1](https://gophp71.org/)

I think PHP7 is the futrue.
Maybe we can release a version to end the PHP5 support?

File: psr_http_server_handler.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#include "config.h"
+#endif
+
+#include "php.h"
+#include "php_psr.h"
+#include "psr_http_server_handler.h"
+
+/* {{{ ReqeustHandlerInterface --------------------------------------------------- */
+
+PHP_PSR_API zend_class_entry * PsrHttpServerReqeustHandlerInterface_ce_ptr;
```
Comment (jbboehr) [inline psr_http_server_handler.c]: `Reqeust` -> `Request`

**Comment by jbboehr** [comment]
While I'd like to keep PHP 5 support until 5.6 EOL (in fact, this repository still supports PHP 5.3), all of the libraries that are tested against have since dropped support for PHP 5, which is why I no longer run the tests under PHP 5 in Travis.

We can easily continue to support PHP 5 via ifdefs around the new classes.

@sergeyklay Do you still need PHP 5 support?

**Comment by jbboehr** [comment]
Otherwise, it looks great!

**Comment by sergeyklay** [comment]
@jbboehr Actually support for PHP 5.x in Phalcon related projects is provided on a best-effort basis. I don't need  PHP 5 support neither for personal projects, nor for public ones. But frankly speaking dropping PHP 5.x support will break backward compatibility and this will require a major release.

**Comment by jbboehr** [comment]
@sergeyklay I've added PHP5 compatibility in a local branch, if you'd like to review: https://github.com/jbboehr/php-psr/tree/psr-15

**Comment by sergeyklay** [comment]
@flyinghail Could you please squash commits into one to keep history clean and avoid future extra work to resolve conflicts on rebase? Let me know if you need instructions.

**Comment by jbboehr** [comment]
@flyinghail You can pull in my commits in the local `psr-15` branch and squash them too, if you'd like.

**Comment by flyinghail** [comment]
@sergeyklay @jbboehr 

I have less experience with "pull requests". If any change is needed, please inform me at any time.

**Pull request closed without merging**